From: Ben Hutchings <bhutchings@solarflare.com>
Date: Wed, 12 Feb 2014 18:59:28 +0000
Subject: [PATCH v2 219/244] sfc: Preserve rx_frm_trunc counters when resizing
 DMA rings
Patch-mainline: v3.15-rc1
Git-commit: 2fa25cf1e1cad74df4b465710e55a62386743614
References: bsc#909618 FATE#317521

We allocate efx_channel structures with kzalloc() so we don't need to
zero-initialise individual fields in efx_probe_channel().  Further,
this function will be called again during DMA ring resizing and we
should not reset any statistics then.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: Shradha Shah <sshah@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/sfc/efx.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/ethernet/sfc/efx.c
+++ b/drivers/net/ethernet/sfc/efx.c
@@ -503,8 +503,6 @@ static int efx_probe_channel(struct efx_
 			goto fail;
 	}
 
-	channel->n_rx_frm_trunc = 0;
-
 	return 0;
 
 fail:
